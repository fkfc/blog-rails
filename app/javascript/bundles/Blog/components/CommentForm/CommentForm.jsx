import React from 'react';
import PropTypes from 'prop-types';

export default function CommentForm(props) {
  const {
    content,
    actionPath,
    authenticityToken,
    method,
  } = props;
  const buttonLabel = (method === 'patch') ? 'Update Comment' : 'Create Comment';

  return (
    <div>
      <form action={actionPath} method="post">
        <input name="_method" readOnly type="hidden" value={method} />
        <input type="hidden" name="authenticity_token" value={authenticityToken} />
        <div className="form-group">
          <textarea name="comment[content]" placeholder="Type your comment here" className="form-control" defaultValue={content} />
        </div>
        <div className="text-right">
          <input type="submit" value={buttonLabel} className="btn btn-primary" />
        </div>
      </form>
    </div>
  );
}

CommentForm.propTypes = {
  content: PropTypes.string,
  actionPath: PropTypes.string,
  authenticityToken: PropTypes.string.isRequired,
  method: PropTypes.string,
};

CommentForm.defaultProps = {
  content: '',
  actionPath: null,
  method: 'post',
};
