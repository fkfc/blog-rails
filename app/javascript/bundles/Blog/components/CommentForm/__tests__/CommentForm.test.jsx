import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import CommentForm from '../CommentForm';

describe('Testing the Post Component', () => {
  configure({ adapter: new Adapter() });
  let props;
  beforeEach(() => {
    props = {
      id: 1,
      content: 'comment content',
      createdAt: 'Aaa, 12 3456',
      postCommentPath: '/postCommentPath',
      authenticityToken: 'token',
    };
  });
  it('Should render correctly', () => {
    const wrapper = shallow(<CommentForm {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
