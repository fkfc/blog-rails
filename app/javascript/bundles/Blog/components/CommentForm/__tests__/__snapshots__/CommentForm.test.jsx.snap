// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Post Component Should render correctly 1`] = `
<div>
  <form
    action={null}
    method="post"
  >
    <input
      name="_method"
      readOnly={true}
      type="hidden"
      value="post"
    />
    <input
      name="authenticity_token"
      type="hidden"
      value="token"
    />
    <div
      className="form-group"
    >
      <textarea
        className="form-control"
        defaultValue="comment content"
        name="comment[content]"
        placeholder="Type your comment here"
      />
    </div>
    <div
      className="text-right"
    >
      <input
        className="btn btn-primary"
        type="submit"
        value="Create Comment"
      />
    </div>
  </form>
</div>
`;
